### PR TITLE
re-add lightweight pfcmd violation management

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Re-adds a lightweight pfcmd violation to open/close/trigger violations
# Impacts

pfcmd
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Re-added support for adding/removing violations through pfcmd

### CODE REVIEW

File: lib/pf/cmd/pf/violation.pm [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        $self->showHelp;
+    }
+    return $EXIT_SUCCESS;
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+Minor parts of this file may have been contributed. See CREDITS.
```
Comment (extrafu) [inline lib/pf/cmd/pf/violation.pm:115]: Please don't put this when it's not the case

File: lib/pf/cmd/pf/violation.pm [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+=cut
+
+sub action_close {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 2 && $self->validate_mac($params[0]) && $self->validate_vid($params[1]) ) {
+        violation_force_close(@params);
+    }
+    else {
+        print STDERR "Insuficent or invalid parameters supplied.\n";
```
Comment (extrafu) [inline lib/pf/cmd/pf/violation.pm:67]: Insufficient

File: lib/pf/cmd/pf/violation.pm [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+=cut
+
+sub action_trigger {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 3 && $self->validate_mac($params[0]) ) {
+        violation_trigger({mac => $params[0], type => $params[1], tid => $params[2]});
+    }
+    else {
+        print STDERR "Insuficent or invalid parameters supplied.\n";
```
Comment (extrafu) [inline lib/pf/cmd/pf/violation.pm:86]: Insufficient

File: lib/pf/cmd/pf/violation.pm [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+=cut
+
+sub action_add {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 2 && $self->validate_mac($params[0]) && $self->validate_vid($params[1]) ) {
+        violation_add(@params);
+    }
+    else {
+        print STDERR "Insuficent or invalid parameters supplied.\n";
```
Comment (extrafu) [inline lib/pf/cmd/pf/violation.pm:105]: Insufficient

File: lib/pf/cmd/pf/violation.pm [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        $self->showHelp;
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:106]: return $EXIT_FAILURE;

File: lib/pf/cmd/pf/violation.pm [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+sub action_trigger {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 3 && $self->validate_mac($params[0]) ) {
+        violation_trigger({mac => $params[0], type => $params[1], tid => $params[2]});
+    }
+    else {
+        print STDERR "Insuficent or invalid parameters supplied.\n";
+        $self->showHelp;
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:87]: return $EXIT_FAILURE;

File: lib/pf/cmd/pf/violation.pm [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+sub action_close {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 2 && $self->validate_mac($params[0]) && $self->validate_vid($params[1]) ) {
+        violation_force_close(@params);
+    }
+    else {
+        print STDERR "Insuficent or invalid parameters supplied.\n";
+        $self->showHelp;
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:68]: return $EXIT_FAILURE;

**Comment by jrouzierinverse** [comment]
for each of the actions you should also create a
parse_<action> method to validate the parameters
returning $TRUE if the args are valid or $FALSE if they are invalid.

sub parse_add {
}

sub parse_close {
}

sub parse_trigger {
}

File: lib/pf/cmd/pf/violation.pm [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+handles 'pfcmd violation add' command
+
+=cut
+
+sub action_add {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 2 && $self->validate_mac($params[0]) && $self->validate_vid($params[1]) ) {
+        violation_add(@params);
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:102]: Validate that this succeeded.

File: lib/pf/cmd/pf/violation.pm [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+handles 'pfcmd violation trigger' command
+
+=cut
+
+sub action_trigger {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 3 && $self->validate_mac($params[0]) ) {
+        violation_trigger({mac => $params[0], type => $params[1], tid => $params[2]});
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:83]: Validate that this succeeded

File: lib/pf/cmd/pf/violation.pm [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+handles 'pfcmd violation close' command
+
+=cut
+
+sub action_close {
+    my ($self) = @_;
+    my @params = $self->action_args;
+    if( @params >= 2 && $self->validate_mac($params[0]) && $self->validate_vid($params[1]) ) {
+        violation_force_close(@params);
```
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/violation.pm:64]: Validate that this succeeded.

**Pull request merged**